### PR TITLE
Added option to change sens in dev console

### PR DESCRIPTION
**Pull request by nukeulater**
- got rid of some compiler warnings in release mode
- moved FOV function and crosshair position change to H2Tweaks 
- moved INI macros to separate file
- removed boolean trap

### CODE REVIEW

**Comment by shayaantx** [comment]
Who tested this?
Why are you changing h2mod return statements?
For everything you changed, please confirm everything still works.

File: xlive/H2MOD.cpp [hunk 1]
```diff
@@ -726,7 +712,7 @@ void __stdcall OnPlayerScore(void* thisptr, unsigned short a2, int a3, int a4, i
 
 #pragma endregion
 
-	return pupdate_player_score(thisptr, a2, a3, a4, a5, a6);
+	pupdate_player_score(thisptr, a2, a3, a4, a5, a6);
```
Comment (shayaantx) [inline xlive/H2MOD.cpp:716]: why no more return here?

File: xlive/H2MOD.cpp [hunk 2]
```diff
@@ -1003,7 +971,7 @@ bool __cdecl OnPlayerSpawn(int a1)
 	//TRACE_GAME("OnPlayerSpawn(a1: %08X)", a1);
 
 	int PlayerIndex = a1 & 0x000FFFF;
-	int ret = pspawn_player(a1);
+	bool ret = pspawn_player(a1);
```
Comment (shayaantx) [inline xlive/H2MOD.cpp]: any reason for this?

File: xlive/H2MOD_Halo2Final.cpp [hunk 3]
```diff
@@ -16,7 +17,8 @@
 */
 
 bool isEnabled = false;
-bool isHosting = false;
+bool H2Server = (h2mod->Server) ? 1 : 0;
```
Comment (shayaantx) [inline xlive/H2MOD_Halo2Final.cpp]: not sure having a bool really helps readability in this case, 

you're just making a new bool of an existing bool.

File: xlive/H2MOD_Halo2Final.h [hunk 4]
```diff
@@ -114,7 +114,7 @@ class Halo2FinalSettings
 class Halo2Final
 {
 public:
-	void Initialize(bool isHost);
+	void Initialize();
```
Comment (shayaantx) [inline xlive/H2MOD_Halo2Final.h:117]: gg

File: xlive/H2Tweaks.cpp [hunk 5]
```diff
@@ -196,3 +196,37 @@ void InitH2Tweaks() {
 void DeinitH2Tweaks() {
 
 }
+
+void setSens(short input_type, float sens) {
+	if (input_type) { //controller
```
Comment (shayaantx) [inline xlive/H2Tweaks.cpp]: how many input types are there?

File: xlive/H2Tweaks.cpp [hunk 6]
```diff
@@ -196,3 +196,37 @@ void InitH2Tweaks() {
...
+		float targetRadiansVehicle = (float)((double)(field_of_view + 10) * 3.14159265f / 180);
+		*(float*)(H2BaseAddr + 0x41D984) = (targetRadians / defaultRadians); //First Person
+		*(float*)(H2BaseAddr + 0x413780) = (targetRadians / defaultRadians); //Vehicle
+	}
+}
+
+void setCrosshairPos(float crosshair_offset) {
+
+	if (!FloatIsNaN(crosshair_offset)) {
+	DWORD CrosshairY = *(DWORD*)((char*)H2BaseAddr + 0x479E70) + 0x1AF4 + 0xf0 + 0x1C;
```
Comment (shayaantx) [inline xlive/H2Tweaks.cpp:228]: needs indent

File: xlive/Read.h [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	}
+
+#define CHECK_ARG_I64(x,y) \
+	if( strstr( str,x ) == str ) \
+	{ \
+		sscanf( str + strlen(x), "%I64x", &y ); \
+		continue; \
+	}
+
+#define gCHECK_ARG(x,y) \
```
Comment (shayaantx) [inline xlive/Read.h]: extra "g" in front of check_arg?

File: xlive/Read.h [hunk 8]
```diff
@@ -0,0 +1,36 @@
+#pragma once
```
Comment (shayaantx) [inline xlive/Read.h]: why is this called "Read"?

Always document header/impl/class files to what they are exactly. In this case, it is for reading arguments, so ReadArgumentUtility seems reasonable.

File: xlive/dllmain.cpp [hunk 9]
```diff
@@ -186,9 +187,9 @@ void trace_game_narrow(LPSTR message, ...)
 
 std::wstring prepareLogFileName(std::wstring logFileName) {
 	std::wstring instanceNumber(L"");
-	if (H2Config_instance_number != 1) {
+	if (H2GetInstanceId() > 1) {
```
Comment (shayaantx) [inline xlive/dllmain.cpp:190]: gg

**Review by shayaantx — CHANGES_REQUESTED** [review]
check my comments

File: xlive/H2MOD.cpp (hunk 1 above)
Comment (nukeulater) [reply xlive/H2MOD.cpp:716]: void

File: xlive/H2MOD.cpp (hunk 2 above)
Comment (NukeULater) [reply xlive/H2MOD.cpp]: i reverted this back

File: xlive/H2MOD_Halo2Final.h (hunk 4 above)
Comment (nukeulater) [reply xlive/H2MOD_Halo2Final.h:117]: :)

File: xlive/H2Tweaks.cpp (hunk 5 above)
Comment (nukeulater) [reply xlive/H2Tweaks.cpp]: eh

File: xlive/H2Tweaks.cpp (hunk 6 above)
Comment (nukeulater) [reply xlive/H2Tweaks.cpp:228]: ok

File: xlive/Read.h (hunk 7 above)
Comment (nukeulater) [reply xlive/Read.h]: thats how it was

File: xlive/Read.h (hunk 8 above)
Comment (nukeulater) [reply xlive/Read.h]: ok will do

File: xlive/dllmain.cpp (hunk 9 above)
Comment (nukeulater) [reply xlive/dllmain.cpp:190]: :)

File: xlive/H2MOD.cpp (hunk 2 above)
Comment (nukeulater) [reply xlive/H2MOD.cpp]: that function returns a bool

File: xlive/H2Tweaks.cpp [hunk 10]
```diff
@@ -198,7 +198,7 @@ void DeinitH2Tweaks() {
 }
 
 void setSens(short input_type, float sens) {
-	if (input_type) { //controller
+	if (input_type == 1) { //controller
```
Comment (shayaantx) [inline xlive/H2Tweaks.cpp]: I asked how many inputs, cause if its just 1 or 0, you just if input == 1 and else for the 0, if not, you don't need to update this really - although its a tad more readable

File: xlive/dllmain.cpp [hunk 11]
```diff
@@ -11,7 +11,7 @@
 #include "H2ConsoleCommands.h"
 #include "H2Config.h"
 #include <sstream>
-#include "Read.h"
+#include "ReadArgumets_INI.h"
```
Comment (shayaantx) [inline xlive/dllmain.cpp]: arguments is spelled wrong, also using _ in header files is ugly

ReadIniArguments

**Review by shayaantx — CHANGES_REQUESTED** [review]
see comments

**Review by shayaantx — APPROVED** [review]

**Pull request merged**